### DISCUSSION TITLE
adding random node to leaf, but after adding new node the zoom level returns to root node,

### DISCUSSION THREAD

**Discussion by senayakagunduz** [d3/d3#3958] [Q&A]
I am using zoomable treemap, for example I added random node to leaf, but after adding new node the zoom level returns to root node, like refreshing the page, how can I prevent this action?

**Comment by Fil** [comment]
It is hard to help with so little information. It probably depends on how you manage the state of your application? (Make sure that changing the data does not reset the view.)
